### DISCUSSION TITLE
Should Rollup not restart when new files are added?

### DISCUSSION THREAD

**Discussion by alexbjorlig** [rollup/rollup#5116] [Q&A]
Hi there, thanks for a great bundler. My objective is to use Rollup for a CLI written in typescript. I run rollup with this command

`rollup -c -w`

When I edit existing files, the watch feature works.

But if I add a new typescript file, I have to manually restart rollup.

Is this expected, or do I miss something obvious?

**Comment by lukastaegert** [comment]
It is rather vague what you mean by "adding". A reproduction would go a long way.

**Reply by alexbjorlig** [reply]
Hi there, thanks for answering.

What I mean by "adding" a new file:
![CleanShot 2023-09-01 at 10 41 46@2x](https://github.com/rollup/rollup/assets/9967422/acb461b6-1d8f-4e25-b0d8-a54e1c77e0c8)

If I should setup a reproduction, do you have some guide or template for node.js/typescript?

**Comment by lukastaegert** [comment]
Basically: Why should Rollup restart when you add a file that is not used by the other files? To use it, you either need to import it, which means changing an existing file, or make it an entry, which means changing the config. "add" does not really describe the details of your workflow.
But in short: Rollup only watches files that are included in the build.

**Comment by alexbjorlig** [comment]
> Basically: Why should Rollup restart when you add a file

Because the new file is a database migration script. When the migrate database command is being run, it will process all scripts in the folder. (but currently, I have to restart rollup).

Do you know of some way this could be possible, so I don't have to restart the rollup process?

**Comment by lukastaegert** [comment]
Write a plugin that manually adds this file as a watch file e.g. in the `buildStart` hook via [`this.addWatchFile`](https://rollupjs.org/plugin-development/#this-addwatchfile)
